core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -42,42 +42,172 @@
+                tok = optTok(lexer);
+            }
+
+            // expect [timestamp(column)]
+
+            ExpressionNode timestamp = parseTimestamp(lexer, tok);
+            if (timestamp != null) {
+                model.setTimestamp(timestamp);
+                tok = optTok(lexer);
+            }
+
+            // expect [latest by]
+
+            if (tok != null && isLatestKeyword(tok)) {
+                parseLatestBy(lexer, model);
+                tok = optTok(lexer);
+            }
+        }
+
+        // expect multiple [[inner | outer | cross] join]
+
+        int joinType;
+        while (tok != null && (joinType = joinStartSet.get(tok)) != -1) {
+            model.addJoinModel(parseJoin(lexer, tok, joinType, masterModel));
+            tok = optTok(lexer);
+        }
+
+        // expect [where]
+
+        if (tok != null && isWhereKeyword(tok)) {
+            ExpressionNode expr = expr(lexer, model);
+            if (expr != null) {
+                model.setWhereClause(expr);
+                tok = optTok(lexer);
+            } else {
+                throw SqlException.$((lexer.lastTokenPosition()), "empty where clause");
+            }
+        }
+
+        // expect [sample by]
+
+        if (tok != null && isSampleKeyword(tok)) {
+            expectTok(lexer, "by");
+            model.setSampleBy(expectLiteral(lexer));
+            tok = optTok(lexer);
+
+            if (tok != null && isFillKeyword(tok)) {
+                expectTok(lexer, '(');
+                do {
+                    final ExpressionNode fillNode = expr(lexer, model);
+                    if (fillNode == null) {
+                        throw SqlException.$(lexer.lastTokenPosition(), "'none', 'prev', 'mid', 'null' or number expected");
+                    }
+                    model.addSampleByFill(fillNode);
+                    tok = tokIncludingLocalBrace(lexer, "',' or ')'");
+                    if (Chars.equals(tok, ')')) {
+                        break;
+                    }
+                    expectTok(tok, lexer.lastTokenPosition(), ',');
+                } while (true);
+                tok = optTok(lexer);
+            }
+        }
+
+        //expect [group by]
+
+        if (tok != null && isGroupKeyword(tok)) {
+            expectTok(lexer, "by");
+            do {
+                tokIncludingLocalBrace(lexer, "literal");
+                lexer.unparse();
+                ExpressionNode n = expr(lexer, model);
+                if (n == null || (n.type != ExpressionNode.LITERAL && n.type != ExpressionNode.CONSTANT)) {
+                    throw SqlException.$(n == null ? lexer.lastTokenPosition() : n.position, "literal expected");
+                }
+
+                model.addGroupBy(n.token);
+
+                tok = optTok(lexer);
+            } while (tok != null && Chars.equals(tok, ','));
+        }
+
+        // expect [order by]
+
+        if (tok != null && isOrderKeyword(tok)) {
+            expectTok(lexer, "by");
```

Comment:
Can we also  put "by" check as sql keyword? Across all clauses pls

Reply:
done